test/html.test.js

```diff
@@ -1073,6 +1086,69 @@ QUnit.test('translates metadata events into WebVTT cues', function() {
   QUnit.equal(cues[2].endTime, mediaSource.duration, 'sourceended is fired');
 });
 
+QUnit.test('cleans up WebVTT cues on sourceclose', function() {
+  let mediaSource = new videojs.MediaSource();
+  let sourceBuffer = mediaSource.addSourceBuffer('video/mp2t');
+
+  mediaSource.duration = Infinity;
+  mediaSource.nativeMediaSource_.duration = 60;
+
+  let addedTracks = [];
+  let removedTracks = [];
+  let metadata = [{
+    cueTime: 2,
+    frames: [{
+      url: 'This is a url tag'
+    }, {
+      value: 'This is a text tag'
+    }]
+  }, {
+    cueTime: 12,
+    frames: [{
+      data: 'This is a priv tag'
+    }]
+  }];
+
+  metadata.dispatchType = 0x10;
+  mediaSource.player_ = {
+    addRemoteTextTrack(options) {
+      let trackEl = {
+        track: {
+          kind: options.kind,
+          label: options.label,
+          addCue(cue) {}
+        }
+      };
+
+      addedTracks.push(trackEl.track);
+      return trackEl;
+    },
+    remoteTextTracks() {
+      return addedTracks;
+    },
+    removeRemoteTextTrack(track) {
+      removedTracks.push(track.kind);
+    }
+  };
+  sourceBuffer.timestampOffset = 10;
+
+  sourceBuffer.transmuxer_.onmessage(createDataMessage('video', new Uint8Array(1), {
+    metadata
+  }));
+  sourceBuffer.transmuxer_.onmessage(createDataMessage('video', new Uint8Array(1), {
```

Comment:
Nah, in fact I should just collapse the two messages into one. No reason to keep both. Thanks for pointing that out.